controllers/pod_node_module_reconciler.go

```diff
@@ -70,8 +72,21 @@ func (pnmr *PodNodeModuleReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	if !podutils.IsPodReady(&pod) || !pod.DeletionTimestamp.IsZero() {
 		logger.Info("Unlabeling node")
 
-		if err := pnmr.deleteLabel(ctx, nodeName, labelName); err != nil {
-			return ctrl.Result{}, fmt.Errorf("could not unlabel node %s: %v", nodeName, err)
+		// Make sure we don't already have a new running pod before unlabeling the node
+		labelSelector := client.MatchingLabels{constants.ModuleNameLabel: moduleName}
```

Comment:
we are supporting both label now, can't we add a pod name as a value to the new label? That way we won't need to query the pods on the node

Reply:
I prefer it as well but @qbarrand prefer to query the pods.

Reply:
IMO no need to expose the pod names, and I think the implementation in this PR looks great.